File: examples/tentacles/hello-world/go/handler.go
package main

import (
	"encoding/json"
	"os"
)

func main() {
	var req struct {
		ID      any            `json:"id"`
		Payload map[string]any `json:"payload"`
	}
	_ = json.NewDecoder(os.Stdin).Decode(&req)
	name := "world"
	if req.Payload != nil {
		if n, ok := req.Payload["name"].(string); ok && n != "" {
			name = n
		}
	}
	out := map[string]any{
		"v": 1, "id": req.ID, "ok": true,
		"result": map[string]any{"arm": "hello-go", "lang": "go", "hello": "hello, " + name},
	}
	_ = json.NewEncoder(os.Stdout).Encode(out)
}

File: examples/tentacles/hello-world/php/handler.php
<?php
$req = json_decode(stream_get_contents(STDIN), true) ?: [];
$name = $req['payload']['name'] ?? 'world';
echo json_encode([
    'v' => 1, 'id' => $req['id'] ?? null, 'ok' => true,
    'result' => ['arm' => 'hello-php', 'lang' => 'php', 'hello' => "hello, $name"],
]);
